### PR TITLE
Fixes memory leaks

### PR DESCRIPTION
**Pull request by bkchr**
Especially the one in sender.c is very important!

### CODE REVIEW

File: picoquic/sender.c [hunk 1]
```diff
@@ -2142,14 +2142,17 @@ int picoquic_prepare_packet(picoquic_cnx_t* cnx,
...
 
             if (ret == 0) {
                 *send_length += segment_length;
                 if (packet->length == 0 ||
                     packet->ptype == picoquic_packet_1rtt_protected_phi0 ||
                     packet->ptype == picoquic_packet_1rtt_protected_phi1) {
+                    if (packet->length == 0) {
+                        free(packet);
+                    }
                     break;
```
Comment (huitema) [inline picoquic/sender.c]: Weird. I am pretty sure I coded exactly that in my recent patch. But you are correct, it did not make it into the PR. Something went wrong. Thanks for fixing it.

File: picoquictest/tls_api_test.c [hunk 2]
```diff
@@ -779,6 +779,8 @@ static int tls_api_one_sim_round(picoquic_test_tls_api_ctx_t* test_ctx,
                     /* useless test, but makes it easier to add a breakpoint under debugger */
                     ret = -1;
                 }
+
+                free(packet);
```
Comment (huitema) [inline picoquictest/tls_api_test.c]: Isn't the data already freed at line 743?

Comment (bkchr) [reply picoquictest/tls_api_test.c]: No, the packet is created in line 762

**Comment by bkchr** [comment]
@huitema I added more memory leak fixes. Some of them are also in the stress test, maybe that helps you with https://github.com/private-octopus/picoquic/issues/206, to reduce the execution time.

**Pull request merged**